test/test_operations.py

```diff
@@ -926,34 +916,38 @@ def func(a, b):
     b = torch.ones([2, 2])
     self.runAtenTest((a, b), func)
 
-  @unittest.skip("Broken by functionalization")
   def test_set(self):
     met.clear_all()
 
     t1 = torch.zeros(50, device=xm.xla_device())
     t1 += 1
     xm.mark_step()
-    self.assertEqual(met.counter_value('DestroyXlaTensor'), 2)
-
-    t1.data = torch.zeros(20, device=xm.xla_device())
     self.assertEqual(met.counter_value('DestroyXlaTensor'), 3)
 
-    t1.set_(torch.zeros(10, device=xm.xla_device()))
+    t2 = torch.zeros(10, device=xm.xla_device())
```

Comment:
lol I am a bit confuse, why is `DestroyXlaTensor` increased when we init the t2?

Reply:
That's a question to @bdhirsh ...

Reply:
if we add debug message in the XLATensor constructor, at least we can know the tensors's shape. I also suspect this is a intermediate tensor.  I am a bit concern about adding test cases that we don't understand waht's the correct output.

Reply:
Okay, let me try to dig around.

Reply:
Apparently, zeros is decomposed into zero_ and thus a the in-place op functionalize pass has introduced a intermediate tensor:
```
[call] op=[aten::zeros], key=[BackendSelect]
  [redispatch] op=[aten::zeros], key=[XLA]
   [call] op=[aten::empty.memory_format], key=[BackendSelect]
    [redispatch] op=[aten::empty.memory_format], key=[XLA]
   [call] op=[aten::zero_], key=[Functionalize]
    [call] op=[aten::zero_], key=[Meta]
    [call] op=[aten::zero], key=[XLA]
     [call] op=[aten::clone], key=[XLA]
     [call] op=[aten::zero_], key=[XLA]
    [call] op=[aten::_propagate_xla_data], key=[XLA]
```